orderapi/src/main/java/fi/hel/verkkokauppa/order/service/pdf/PdfGenerationService.java

```diff
@@ -0,0 +1,108 @@
+import fi.hel.verkkokauppa.common.rest.RestServiceClient;
+import fi.hel.verkkokauppa.common.rest.dto.configuration.MerchantDto;
+import fi.hel.verkkokauppa.common.rest.dto.payment.PaymentDto;
+import fi.hel.verkkokauppa.order.api.data.OrderAggregateDto;
+import fi.hel.verkkokauppa.order.api.data.OrderItemDto;
+import fi.hel.verkkokauppa.order.model.pdf.GenerateOrderConfirmationPDFRequestDto;
+import fi.hel.verkkokauppa.order.service.order.OrderService;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class PdfGenerationService {
+
+    private final Logger log = LoggerFactory.getLogger(PdfGenerationService.class);
+
+
+    @Autowired
+    private RestServiceClient restServiceClient;
+
+    @Autowired
+    private OrderService orderService;
+
+    @Autowired
+    CommonServiceConfigurationClient commonServiceConfigurationClient;
+
+    @Value("${payment.service.url:http://payment-api:8080}")
+    private String paymentServiceUrl;
+
+    @Value("${merchant.service.url:http://merchant-api:8080}")
+    private String merchantServiceUrl;
+
+
+
+    public GenerateOrderConfirmationPDFRequestDto getPDFRequestDto(String orderId) throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+        GenerateOrderConfirmationPDFRequestDto dto = new GenerateOrderConfirmationPDFRequestDto();
+        dto.setOrderId(orderId);
+
+        try {
+
+            // get Order
+            OrderAggregateDto orderDto = orderService.orderAggregateDto(orderId).getBody();
+
+            // check that there are items
+            List<OrderItemDto> items = orderDto.getItems();
+            if (items == null || items.isEmpty()) {
+                throw new Exception("Order has not items");
+            }
+
+            dto.setItems(items);
+            dto.setCustomerFirstName(orderDto.getOrder().getCustomerFirstName());
+            dto.setCustomerLastName(orderDto.getOrder().getCustomerLastName());
+            dto.setCustomerEmail(orderDto.getOrder().getCustomerEmail());
+
+            // Get Payment
+            JSONObject paymentResponse = restServiceClient.makeAdminGetCall(paymentServiceUrl + "/payment-admin/online/get?orderId=" + orderId);
+            PaymentDto paymentDto = objectMapper.readValue(paymentResponse.toString(), PaymentDto.class);
+            dto.setPayment(paymentDto);
+
+            // Get merchant
+            String merchantId = orderDto.getItems().get(0).getMerchantId();
+            String nameSpace = orderDto.getOrder().getNamespace();
+            MerchantDto merchantDto = commonServiceConfigurationClient.getMerchantModel(merchantId, nameSpace);
+
+            merchantDto.getConfigurations().forEach(configuration -> {
```

Comment:
Should we wrap some of these to new functions like add merchantDatatoPdfDto etc, those can be private function on this class and makes this somewhat smaller one.